### PR TITLE
Refine filter parser

### PR DESCRIPTION
**Pull request by Lloyd-Pottiger**
### What problem does this PR solve?

Issue Number: ref #6233 

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp [hunk 1]
```diff
@@ -431,27 +377,22 @@ RSOperatorPtr FilterParser::parseRFInExpr(
...
         if (!isColumnExpr(target_expr))
-        {
             return createUnsupported(target_expr.ShortDebugString(), "rf target expr is not column expr", false);
-        }
         auto column_define = cop::getColumnDefineForColumnExpr(target_expr, columns_to_read);
-        attr = Attr{.col_name = column_define.name, .col_id = column_define.id, .type = column_define.type};
-        std::for_each(setElements.begin(), setElements.end(), [&](Field element) { values.push_back(element); });
+        auto attr = Attr{.col_name = column_define.name, .col_id = column_define.id, .type = column_define.type};
+        // FIXME: for timestamp literal, we should convert it to UTC timezone
+        Fields values(setElements.begin(), setElements.end());
```
Comment (Lloyd-Pottiger) [inline dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp:389]: https://github.com/pingcap/tiflash/issues/8222

**Review by hongyunyan — APPROVED** [review]

**Comment by Lloyd-Pottiger** [comment]
unstable test fullstack-test2/ddl/alter_partition_by.test

**Pull request merged**